platforms/allwinner-d1/core/src/ccu.rs

```diff
@@ -0,0 +1,272 @@
+//! This module provides a higher-level interface for the Clock Controller Unit (CCU).
+use d1_pac::CCU;
+use d1_pac::DMAC;
+use d1_pac::{SMHC0, SMHC1, SMHC2};
+use d1_pac::{SPI0, SPI_DBI};
+use d1_pac::{TWI0, TWI1, TWI2, TWI3};
+use d1_pac::{UART0, UART1, UART2, UART3, UART4, UART5};
+
+pub struct Ccu {
+    ccu: CCU,
+}
+
+/// Trait to be implemented for module clocks that can be gated and reset
+pub trait BusGatingResetRegister {
+    /// Enable or disable the clock reset bit
+    fn gating(ccu: &mut CCU, pass: bool);
+    /// Enable or disable the clock gating bit
+    fn reset(ccu: &mut CCU, deassert: bool);
+}
+
+// TODO: should this move into the `Clint`?
+fn sdelay(delay_us: usize) {
+    let clint = unsafe { crate::clint::Clint::summon() };
+    let t = clint.get_mtime();
+    // TODO: confirm clock source of mtime
```

Comment:
mtime is sourced from the CPU core clock. It will tick at the same rate of the CPU core. This means it will also be affected by the clock selected for the CPU.

We could rename this `rename_jiffies`, if we want to adopt the kernel naming style.

Reply:
Did you mean something like `delay_jiffies`?

Reply:
Ah yes, sorry :)

Reply:
> mtime is sourced from the CPU core clock. It will tick at the same rate of the CPU core. This means it will also be affected by the clock selected for the CPU.

Btw, did you find this documented somewhere? No matter what search terms I'm using, 
I can't find any useful info about how the C906 CLINT increments mtime :sweat_smile:

Reply:
So, I'm not actually sure if mtime's clint is different than "the riscv mtime register", and it seems I was actually also wrong here 🤦:

https://five-embeddev.com/riscv-isa-manual/latest/machine.html#machine-timer-registers-mtime-and-mtimecmp

> Platforms provide a real-time counter, exposed as a memory-mapped machine-mode read-write register, mtime. mtime must increment at constant frequency, and the platform must provide a mechanism for determining the period of an mtime tick. The  mtime register will wrap around if the count overflows.